messaging/src/main/java/org/axonframework/messaging/unitofwork/UnitOfWork.java

```diff
@@ -65,34 +71,71 @@ public class UnitOfWork implements ProcessingLifecycle {
     private final String identifier;
     private final UnitOfWorkProcessingContext context;
 
-    /**
-     * Constructs a {@code UnitOfWork} with a {@link UUID#randomUUID() random UUID String}. Will execute provided
-     * actions on the same thread invoking this Unit of Work.
-     */
-    public UnitOfWork() {
-        this(UUID.randomUUID().toString());
-    }
-
-    /**
-     * Constructs a {@code UnitOfWork} with the given {@code identifier}. Will execute provided actions on the same
-     * thread invoking this Unit of Work.
-     *
-     * @param identifier The identifier of this Unit of Work.
-     */
-    public UnitOfWork(String identifier) {
-        this(identifier, Runnable::run);
-    }
-
     /**
      * Constructs a {@code UnitOfWork} with the given {@code identifier}, processing actions through the given
      * {@code workScheduler}.
      *
      * @param identifier    The identifier of this Unit of Work.
      * @param workScheduler The {@link Executor} used to process the steps attached to the phases in this Unit of Work
      */
-    public UnitOfWork(String identifier, Executor workScheduler) {
+    @Internal
+    UnitOfWork(@Nonnull String identifier, @Nonnull Configuration configuration) {
+        Objects.requireNonNull(identifier, "identifier may not be null");
+        Objects.requireNonNull(configuration, "configuration may not be null");
         this.identifier = identifier;
-        this.context = new UnitOfWorkProcessingContext(identifier, workScheduler);
+        this.context = new UnitOfWorkProcessingContext(
+                identifier,
+                configuration.workScheduler,
+                configuration.applicationContext
+        );
+    }
+
+    /**
+     * Configuration record for {@link UnitOfWork} construction parameters.
+     * <p>
+     * Defines the work scheduler and application context used during unit of work processing.
+     *
+     * @param workScheduler      The {@link Executor} for processing unit of work actions.
+     * @param applicationContext The {@link ApplicationContext} for component resolution.
+     */
+    public record Configuration(
```

Comment:
Not sure I like this inner `Configuration` object for something so rudimentary as the `UnitOfWork`...do we really need it?

Reply:
Sounds better already, I'll review it soon.